src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.TextChunk.cs

```diff
@@ -23,22 +25,22 @@ private struct TextChunk : IDisposable
             /// capitalized runs and lowercase runs.  i.e. if you have AAbb, then there will be two 
             /// character spans, one for AA and one for BB.
             /// </summary>
-            public readonly StringBreaks CharacterSpans;
+            public readonly ArrayBuilder<TextSpan> CharacterSpans;
```

Comment:
No need to encode the stringbreaks (camel humps) for the pattern.  Encoding is to save space, at the cost of extra CPU.  However, we only need a single allocation for the humps for the pattern.  But we need to look at the humps for every string we compare against.  So we def want to optimize for CPU or memory here.